nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/s3/PutS3ObjectMultipart.java

```diff
@@ -0,0 +1,550 @@
+
+    public static final long MIN_BYTES_INCLUSIVE = 50L * 1024L * 1024L;
+    public static final long MAX_BYTES_INCLUSIVE = 5L * 1024L * 1024L * 1024L;
+    public static final String PERSISTENCE_ROOT = "conf/state/";
+
+    public static final PropertyDescriptor EXPIRATION_RULE_ID = new PropertyDescriptor.Builder()
+            .name("Expiration Time Rule")
+            .required(false)
+            .expressionLanguageSupported(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor STORAGE_CLASS = new PropertyDescriptor.Builder()
+            .name("Storage Class")
+            .required(true)
+            .allowableValues(StorageClass.Standard.name(), StorageClass.ReducedRedundancy.name())
+            .defaultValue(StorageClass.Standard.name())
+            .build();
+
+    public static final PropertyDescriptor PART_SIZE = new PropertyDescriptor.Builder()
+            .name("Part Size")
+            .description("Specifies the Part Size to be used for the S3 Multipart Upload API.  The flow file will be " +
+                    "broken into Part Size chunks during upload.  Part size must be at least 50MB and no more than " +
+                    "5GB, but the final part can be less than 50MB.")
+            .required(true)
+            .defaultValue("5 GB")
+            .addValidator(StandardValidators.createDataSizeBoundsValidator(MIN_BYTES_INCLUSIVE, MAX_BYTES_INCLUSIVE))
+            .build();
+
+    public static final List<PropertyDescriptor> properties = Collections.unmodifiableList(
+            Arrays.asList(KEY, BUCKET, PART_SIZE, ENDPOINT_OVERRIDE, ACCESS_KEY, SECRET_KEY, CREDENTIALS_FILE,
+                    SSL_CONTEXT_SERVICE, STORAGE_CLASS, REGION, TIMEOUT, EXPIRATION_RULE_ID,
+                    FULL_CONTROL_USER_LIST, READ_USER_LIST, WRITE_USER_LIST, READ_ACL_LIST, WRITE_ACL_LIST, OWNER));
+
+    final static String S3_BUCKET_KEY = "s3.bucket";
+    final static String S3_OBJECT_KEY = "s3.key";
+    final static String S3_UPLOAD_ID_ATTR_KEY = "s3.uploadId";
+    final static String S3_VERSION_ATTR_KEY = "s3.version";
+    final static String S3_ETAG_ATTR_KEY = "s3.etag";
+    final static String S3_EXPIRATION_ATTR_KEY = "s3.expiration";
+    final static String S3_USERMETA_ATTR_KEY = "s3.usermetadata";
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return properties;
+    }
+
+    @Override
+    protected PropertyDescriptor getSupportedDynamicPropertyDescriptor(final String propertyDescriptorName) {
+        return new PropertyDescriptor.Builder()
+                .name(propertyDescriptorName)
+                .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+                .expressionLanguageSupported(true)
+                .dynamic(true)
+                .build();
+    }
+
+    protected File getPersistenceFile() {

```

Comment:
I guess, the same comment about _private_ would apply here and in other similar methods